agent/status.go

```diff
@@ -105,6 +108,35 @@ func (a *Agent) statusHandler() http.HandlerFunc {
 	}
 }
 
+type TriggerEndTaskResp struct {
+	apimodels.TaskEndDetail `json:",omitempty"`
```

Comment:
This is a pretty big struct, and contains a bunch of things we don't need. I would suggest letting the user set only the fields we're interseted in and then converting to another struct if you need to.

Reply:
Yeah I was torn between just using this struct or putting the things we needed directly into TriggerEndTaskResp. I thought that using TaskEndDetail would result in less refactoring in the future. Right now we're only allowing description, status, and failure type though.